### PR TITLE
feat(core): report how many places each declaration examined

### PR DESCRIPTION
**Pull request by oekazuma**
## Why

**Verifying a real project meant planting a deliberate violation to see whether anything fired.**

`architecture/reserved-name-placement` is glob-configured and emits no pass results, so a compliant project produces zero routes, zero findings and zero diagnostics. Confirming that `0` meant "the tree complies" rather than "nothing was checked" required adding a violating directory, observing the finding, and deleting it again. Nothing in the output could answer the question.

The charter's inverse-precision gate names this exactly —

> zero findings reads identically as "the project complies" and "the declaration matched nothing"

— and records the fix as its own unbuilt spec. This is that spec, one level finer: per **declaration**, not per rule. One configuration of this rule carries eight declarations, and "the rule examined 137 places" does not answer "did `parts` see 28 or 0?".

**The gap got wider in 0.42.0, not narrower.** Before it, one of the rule's diagnostics fired on a correct declaration — and that false note was the only evidence the rule had run. Fixing it left a compliant project with a completely empty result. The observability was accidental, and correcting the diagnostic removed it.

**More diagnostics cannot answer this.** Reporting a declaration that examined nothing is precisely the case 0.42.0 deliberately made silent: a convention document declaring every permitted position, including ones nothing occupies yet, is correct, and telling its author to "remove the declaration" is advice to delete a check they will want. What a reader needs is information, not a verdict.

## What

A top-level `examined` map in the JSON report — rule id, then declaration label, then how many places that declaration judged.

```jsonc
"examined": {
  "architecture/reserved-name-placement": {
    "capitalisedUnitPlacements.parts → src/**": 28,
    "anyCaseUnitPlacements.types → src/**": 0
  }
}
```

**The engine owns the sink.** `runRules` builds it and returns `{ results, examined }`, rather than each of the three call sites threading one. A caller that forgot would drop the counts **silently** — the exact failure this feature removes, and a shape this repository hit twice recently in option forwarding.

**Not on `rules[id]`, and the reporters guide already said why:** "The counts describe the report, not the tree." `findings`/`passed` describe what survived reporting; this describes what the analysis examined and is deliberately unfiltered by `--diff`, `--baseline` and suppressions. Two scopes behind sibling keys with nothing marking the difference is one field carrying two meanings — the shape behind two defects fixed earlier this week. A top-level map makes the difference structural, next to `inventories`, which is there for the same reason. The guide's sentence is now scoped to `rules`.

**The labels are the diagnostic's own strings**, verbatim. The count is what makes a *silent* declaration legible beside the diagnostic that describes a *broken* one; different names for the same declaration would defeat that.

**Three states, deliberately:** no entry (the rule doesn't count, or returned early), an empty entry (it counts and nothing is declared), and an entry containing `0` (a declaration judged nothing).

### Zero means the declaration judged nothing, and nothing more

An earlier draft of the design claimed zero meant "live, reachable and currently unoccupied". Review falsified that by execution with two supported configurations in which zero appears while occupying directories exist: a sibling map's empty value ungoverns the name everywhere, and an `overrides`-supplied `exclude` skips the directories while the diagnostic classifier consults only the global `exclude` — producing a zero with **no diagnostic at all**.

The rule reaches its judging phase past five early exits, each a separate reason a count can be zero. The design enumerates them; the report states a number and stops. **The useful direction is sound and is the one documented: a non-zero count is what distinguishes "complies" from "nothing was checked".**

## Verification

`core` 1295, `cli` 838, `vite` 207 pass; `tsc --noEmit` clean in all three; lint and format clean across 969 files; docs gates 27; `floor-smoke` 8/8.

The whole-branch review ran **13 mutations**, seven behavioural probes and an end-to-end run of the built CLI; the fix-wave re-review added four more mutations proving the new tests load-bearing, and re-ran all five zero-causes and four counting scenarios.

## The interesting part: the feature's only visible surface was pinned by nothing

The final review deleted the single line that puts `examined` into the report and ran everything: **2333 tests passed with the field gone from every report.** Deleting the argument at either report-producing caller left those suites green too. Both trailing parameters of `buildJsonReport` are optional, so the type system did not catch it either — "the engine owns it, so omission is a type error" held at the `runRules` boundary but not at the reporter boundary, which is the one a user's report actually crosses.

The design's testing list required exactly this ("the two report-producing callers carry the counts, **asserted** by enumerating the call sites, not by sampling"). **The plan's own self-review downgraded it to a grep** — a one-off manual command, not an assertion. A feature built to stop a silent absence shipped its own silent absence, one review away from merging.

Two counting defects were found the same way. A duplicated glob inside one value multiplied the count — `parts: 'src/lib/** | src/routes/** | src/lib/**'` would report 56 where 28 were judged, with no diagnostic, because the duplicate deduped into the alternatives map and counted as used. And an `overrides` layer that **replaces** a global declaration's value inflated that global declaration's count, so a single report claimed the same label had both judged a place and matched no directory — breaking the label join the design calls its whole justification. The increment now fires only where the *resolved* value judges.

Earlier in execution, two of the five early exits turned out to be pinned by nothing: the increment could move above the exclusion check or the root check with the suite green — and the exclusion exit is the one the design singles out in *Deliberately not solved* as a deliberate, documented contract.

**The recurring shape is worth naming.** Four of the plan's own instructions were wrong and each was caught by executing them: a caller grep scoped to `src/` that missed a test file (in a plan whose point is enumerating callers), a fixture asserting `0` for a directory the spec says is judged, a mutation row its own named test could not catch, and the self-review above. Every one surfaced because the process runs each new test against the unchanged code first and works a mutation table where each row names the single test that must fail.

Design: `docs/superpowers/specs/2026-08-07-examined-counts-design.md`. Plan: `docs/superpowers/plans/2026-08-07-examined-counts.md`, corrected in place each time execution proved a step wrong.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * JSON reports now include per-rule, per-declaration examination counts, including zero-count entries.
  * Analysis results expose unfiltered counts, helping distinguish inspected locations from reported findings.
* **Documentation**
  * Added guidance explaining examination counts and how filtering, suppression, baselines, diffs, and overrides affect reported results.
* **Tests**
  * Added coverage for count accuracy, zero-count scenarios, duplicate patterns, and JSON serialization across CLI and build analyses.

### CODE REVIEW

File: docs/src/content/docs/guides/(reporting)/reporters.md [hunk 1]
```diff
@@ -123,14 +128,26 @@ level (`--ignore`, `--rules`, `--category`, or `rules: { id: 'off' }` in config)
...
+that declaration judged, whether it permitted them or rejected them. It is not narrowed by `--diff`,
+`--baseline` or suppressions: those apply to `rules`, and `examined` describes the analysis rather than the
+report, which is why it sits beside `rules` instead of inside it. There are three states: a rule that reports
+no counts has no entry; a rule that counts but whose configuration declares nothing has an empty entry —
+enabling `architecture/reserved-name-placement` in top-level `rules` without declaring a placement reaches
+exactly that; and a declaration that judged nothing is present with `0`. The entries that do appear use the
+exact declaration label the rule's own diagnostic names —
+`architecture/reserved-name-placement`'s aggregated finding and its `examined` entry share the string
+`capitalisedUnitPlacements.parts → src/**`, so the two can be read together. A non-zero count is what tells
+you the declaration was checked at all — the question a rule with no findings otherwise leaves open.
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/guides/(reporting)/reporters.md:150]: Added in both guides: the map itself is optional — a run in which no rule reported counts omits `examined` entirely — stated before the three entry-level states so the field-level and entry-level absences stay distinct. `buildJsonReport` includes the key only when the outer map is non-empty, so a consumer must treat the whole map as optional, not just its entries.
